File: Practicum/MngClient/src/app/models/employeePost.model.ts
import { RoleEmployee } from "./roleEmployee.model";

export class EmployeePost{
    
    
    firstName!: string;
    lastName!: string;
    tz!: string;
    dateOfBirth!: Date;
    beginningOfWork!: Date;
    gender!: boolean;
    employeeRoles!: RoleEmployee[];

    constructor( firstName: string, lastName: string, tz: string, dateOfBirth: Date, beginningOfWork: Date, gender: boolean, 
        employeeRoles: RoleEmployee[]){
           
            this.firstName = firstName;
            this.lastName = lastName;
            this.tz = tz;
            this.dateOfBirth = dateOfBirth;
            this.beginningOfWork = beginningOfWork;
            this.gender = gender;
            this.employeeRoles = employeeRoles;
        }
}

File: Practicum/MngServer/Mng.Api/Models/EmployeePostModel.cs
using Mng.Core.Entities;
using Mng.Data.DTO;

namespace Mng.Api.Models
{
    public class EmployeePostModel
    {
        public string FirstName { get; set; }

        public string LastName { get; set; }

        public string TZ { get; set; }

        public DateTime DateOfBirth { get; set; }

        public Gender Gender { get; set; }

        public DateTime BeginningOfWork { get; set; }

        public bool ActivityStatus { get; set; }

        public List<int> EmployeeRoles { get; set; }
    }
}
